samples/helpers/ColorConsoleLogger/ColorConsoleLogger.cs

```diff
@@ -57,23 +57,20 @@ public bool IsEnabled(LogLevel logLevel)
         /// <param name="formatter">The formatter to be used for formatting the log message.</param>
         public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
         {
-            lock (s_lockObject)
```

Comment:
We were locking too early.

Also, the sample was stopping, and it looked like a deadlock.

I moved the lock lower and cut out one feature (multiple colors on the same line) and have not witnessed the deadlock since.

Reply:
I don't have much knowledge on the ColorConsoleLogger so this question might not make sense, but would there be a way to move the operation somewhere else instead of getting rid of it altogether?

Reply:
I am uncertain why moving the lock change the behavior, but if that is the case, one of the console logging step may have issue during testing?

Reply:
No chance to move elsewhere. We aren't getting rid of the logging, just the distinct color for the timestamp prefix on each message.